### ISSUE TITLE
Errors do not seem to propagate correctly

### ISSUE THREAD

**Issue by airdrummingfool** [MadisonReed/square-connect-api#6]
I've been trying to make requests using some `v2` API methods, and each time I do, the response is an empty error object (`{}`) and an `undefined` response. I added some debug printing, and Square is sending back meaningful errors; it appears that they just aren't coming through the library to my code.

Examples (from `handleRequest`, printed on line 92 of `squareConnect.js`):
`getCustomer`:
* `err`: `null`
* `response`: ```{"statusCode":404,"body":"{\"errors\":[{\"category\":\"INVALID_REQUEST_ERROR\",\"code\":\"NOT_FOUND\",\"detail\":\"API endpoint for URL path `/v2/customers/` and HTTP method `GET` is not found.\"}]}","headers":{"content-type":"application/json","vary":"Origin, Accept-Encoding","x-content-type-options":"nosniff","x-download-options":"noopen","x-frame-options":"SAMEORIGIN","x-permitted-cross-domain-policies":"none","x-xss-protection":"1; mode=block","date":"Fri, 06 Jan 2017 08:02:15 GMT","connection":"close","strict-transport-security":"max-age=631152000","content-length":"158"},"request":{"uri":{"protocol":"https:","slashes":true,"auth":null,"host":"connect.squareup.com","port":443,"hostname":"connect.squareup.com","hash":null,"search":null,"query":null,"pathname":"/v2/customers/","path":"/v2/customers/","href":"https://connect.squareup.com/v2/customers/"},"method":"GET","headers":{"Authorization":"Bearer sandbox-sq0atb-AoWNjJLmoIIn2IS6NE6o1g","Accept":"application/json","User-Agent":"square-connect-api"}}}```
* `body`: ```"{\"errors\":[{\"category\":\"INVALID_REQUEST_ERROR\",\"code\":\"NOT_FOUND\",\"detail\":\"API endpoint for URL path `/v2/customers/` and HTTP method `GET` is not found.\"}]}"```

`createCustomer`:
* `err`: `null`
* `response`: ```{"statusCode":400,"body":{"errors":[{"category":"INVALID_REQUEST_ERROR","code":"BAD_REQUEST","detail":"At least one of the `given_name`, `surname`, `company_name`, `email_address`, or `phone_number` is required for a customer."}]},"headers":{"content-type":"application/json","vary":"Origin, Accept-Encoding","x-content-type-options":"nosniff","x-download-options":"noopen","x-frame-options":"SAMEORIGIN","x-permitted-cross-domain-policies":"none","x-xss-protection":"1; mode=block","date":"Fri, 06 Jan 2017 08:02:15 GMT","connection":"close","strict-transport-security":"max-age=631152000","content-length":"205"},"request":{"uri":{"protocol":"https:","slashes":true,"auth":null,"host":"connect.squareup.com","port":443,"hostname":"connect.squareup.com","hash":null,"search":null,"query":null,"pathname":"/v2/customers","path":"/v2/customers","href":"https://connect.squareup.com/v2/customers"},"method":"POST","headers":{"Authorization":"Bearer sandbox-sq0atb-AoWNjJLmoIIn2IS6NE6o1g","Accept":"application/json","User-Agent":"square-connect-api","content-type":"application/json","content-length":50}}}```
* `body`: ```{"errors":[{"category":"INVALID_REQUEST_ERROR","code":"BAD_REQUEST","detail":"At least one of the `given_name`, `surname`, `company_name`, `email_address`, or `phone_number` is required for a customer."}]}```

It looks like `handleError` (line 109 of `squareConnect.js`) is referencing `response.statusMessage` to create the `errInfo` object, however there is no `statusMessage` property on either one of the `response`s I received from Square.

**Comment by cmswalker** [comment]
@airdrummingfool will have this fixed by end of today. Thanks for finding

**Comment by cmswalker** [comment]
@airdrummingfool So I checked this out and all looks fine. The request and responses for the customer endpoints are working as expected.  Would you mind posting how exactly you are instantiating the square client and calling the methods?  I used a valid location/access token to instantiate and a valid customer Id for getting and all was ok.  For obvious reasons, obfuscate your keys when you post them here, or you can email me directly with your example.

**Comment by cmswalker** [comment]
Its also worth noting that square's API has timeout issues where it simply hangs and never returns a status code or error. I've run into this quite a bit when trying to process multiple transactions async. Instead of throttling appropriately, they simply timeout and the connection is hung.  So if you're doing multiple processes in parallel, I'd try to limit them to 10 at a time for starters.

**Comment by airdrummingfool** [comment]
I'm not running this in production yet, so I'm not trying more than 1 or two requests per minute.

It may be important to note that when I perform the requests with good data, the responses work correctly. The issue I'm having is that **if there is an error in my request (Square can't find a customer, I don't provide good data, etc), that the errors do not come through** to me. This is important because I rely on errors at certain points in my application.

Here are the details of my usage:
```js
var SquareSandbox = new SquareConnect([location ID], [access Token], true);

// to cause the first error, I call getCustomer with an empty string for customer ID
SquareSandbox.getCustomer("", function(err, response) {
    // err = {}
    // response = undefined
});
// When I print the values received in handleRequest line 91 (err, response, body), they are as listed in my first post

// To cause the second error, I call createCustomer without required data
var customerData = { nickname: "Tommy" };
SquareSandbox.createCustomer(customerData, function(err, resposne) {
    // err = {}
    // response = undefined
});
// When I print the values received in handleRequest line 91 (err, response, body), they are as listed in my first post
```

Let me know if there is other information I can provide that would be helpful.

**Comment by cmswalker** [comment]
If i run the exact same thing for getCustomer I get 

```
{"statusCode":404,"message":"Not Found","body":"{\"errors\":[{\"category\":\"INVALID_REQUEST_ERROR\",\"code\":\"NOT_FOUND\",\"detail\":\"API endpoint for URL path `/v2/customers/` and HTTP method `GET` is not found.\"}]}"}
```

**Comment by cmswalker** [comment]
and if i run the exact same thing for createCustomer I get

```
err Error: {"statusCode":400,"message":"Bad Request","body":{"errors":[{"category":"INVALID_REQUEST_ERROR","code":"BAD_REQUEST","detail":"At least one of the `given_name`, `surname`, `company_name`, `email_address`, or `phone_number` is required for a customer."}]}}
```

**Comment by cmswalker** [comment]
what version of node are you running?

**Comment by airdrummingfool** [comment]
6.9.1. I'll re-run the code on my end and see if there's something I'm missing. Thanks for your help!

**Comment by airdrummingfool** [comment]
AH. It appears that `JSON.stringify()` returns `{}` when given an `Error` object. I had assumed that the `err` parameter was a simple POJO, though once I started looking through the code I did see that it was actually an `Error`. I didn't know that `stringify()` didn't work in that case.

It might be helpful to indicate in the README that the `err` parameter is an `Error` object, although that might be a convention I'm unaware of (I'm new to Node 6).

Thanks again for your help!
